File: problems/6/solution.cpp
ListNode* Solution::removeNthNodeFromList(ListNode* head, int n) {
	if (!head) {
		return NULL;
	}

    ListNode* fast = head;
    ListNode* slow = head;

    for(int i = 0; i < n; i++) {
        fast = fast->next;
    }

    //if remove the first node
    if (fast == NULL) {
        head = head->next;
        return head;
    }

    while (fast->next != NULL) {
        fast = fast->next;
        slow = slow->next;
    }

    slow->next = slow->next->next;

    return head;
}

File: problems/6/solution.java
// ListNode type
// class ListNode {
//     int val;
//     ListNode next;
//     ListNode(int x) { val = x; next = null; }
// }

class Solution {
    public static ListNode removeNthNodeFromList(ListNode head, int n) {
        if (head == null) {
            return null;
        }

        ListNode fast = head;
        ListNode slow = head;

        for (int i = 0; i < n; i++) {
            fast = fast.next;
        }

        //if remove the first node
        if (fast == null) {
            head = head.next;
            return head;
        }

        while (fast.next != null) {
            fast = fast.next;
            slow = slow.next;
        }

        slow.next = slow.next.next;
        return head;
    }
}
